analyses/tumor-purity-exploration/01_explore-tumor-purity.Rmd

```diff
@@ -0,0 +1,503 @@
+  ggtitle("Cancer group display after filtering")
+```
+
+
+### Thresholding within groups
+
+Rather than considering a single global threshold, how would this look if we did a _per group_ threshold?
+Specifically, let's see how this looks if we use the _per cancer-group median_ as the threshold and filter from there.
+In addition, we may wish to impose a global minimum to ensure that all samples that at least some baseline tumor fraction level.
+
+```{r}
+# Ensure that all samples have at least 0.7 tumor purity
+global_minimum <- 0.7
+
+cancer_group_thresholds_df <- rna_stranded_df %>%
+  group_by(cancer_group_display) %>%
+  summarize(cancer_group_tumor_threshold = median(tumor_fraction), 
+            n_tumors = n()) 
+
+# Just something to count up the total bh for later division
+total_bh_df <- rna_stranded_df %>%
+  count(broad_histology_display, name = "n_tumors")
+```
+
+Now we can see how many samples would be retained:
+
+```{r}
+rna_stranded_df_thresholded <- rna_stranded_df %>%
+  inner_join(cancer_group_thresholds_df) %>%
+  # Pass both thresholds
+  filter(tumor_fraction >= global_minimum, 
+         tumor_fraction >= cancer_group_tumor_threshold) %>%
+  unique() %>%
+  # bring back the sample_id which may be needed in other contexts
+  inner_join(
+    select(metadata, 
+           Kids_First_Biospecimen_ID, 
+           sample_id)
+  )
+
+# How many cancer groups?
+remaining_cg <- rna_stranded_df_thresholded %>%
+  count(cancer_group_display, name = "count_remaining") %>%
+  # join again to get n_tumors
+  inner_join(cancer_group_thresholds_df) %>%
+  mutate(percent_remaining = count_remaining/n_tumors) 
+remaining_cg
+
+# How many broad histologies groups?
+remaining_bh <- rna_stranded_df_thresholded %>%
+  count(broad_histology_display, name = "count_remaining") %>%
+  # join again to get n_tumors
+  inner_join(total_bh_df) %>%
+  mutate(percent_remaining = count_remaining/n_tumors) 
+remaining_bh
+```
+
+
+```{r}
+plot_remaining_frac2 <- function(df, var, title, fill) {
+  ggplot(df, 
+         aes(x = {{var}}, y = percent_remaining)) + 
+    geom_col(fill = fill, color = "black" ) + 
+    geom_text(aes(y = percent_remaining+0.025, label = count_remaining), 
+              color = "red") +
+    # guide line at y=0.5
+    geom_hline(yintercept = 0.5, alpha = 0.7) +
+    labs(
+      y = "Percent of samples that remain after thresholding",
+      title = title, 
+      subtitle = "text labels are remaining counts"
+    ) +
+    theme(axis.text.x = element_text(angle = 30, size = 8, hjust = 1))
+}
+
+plot_remaining_frac2(remaining_cg, cancer_group_display, "Remaining cancer groups", "darkolivegreen1")
+plot_remaining_frac2(remaining_bh, broad_histology_display, "Remaining broad histology groups", "darkolivegreen4")
+```
+
+
+Let's also export this thresholded version of the data frame.
+
+```{r}
+readr::write_tsv(rna_stranded_df_thresholded, 
+                 rna_stranded_thresholded_df_file)
```

Comment:
Only the IDs please! Or IDs and tumor fraction. If the other fields changed because of changes to the palette or histologies file, someone could be working off of outdated information in this file alone. Hopefully this doesn't matter in practice because those things don't change, but as a matter of practice...